### PR TITLE
fix(TUP-21073): [GuessSchema] Guess Schema process fails in case of messages in error stream

### PR DESCRIPTION
**Pull request by cmeng-talend**
**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://jira.talendforge.org/browse/TUP-21073

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/messages.properties [hunk 1]
```diff
@@ -24,6 +24,7 @@ TaCoKitConfiguration.wizard.exception.message.default=Please see details below.
 TaCoKitConfiguration.composite.exception.message.default=An error happened during the operation
 guessSchema.button=Guess {0} Schema
 guessSchema.error=Error ocurred during guessing schema.
+guessSchema.error.empty=Guess schema failed, no error message returned from the guess schema execution.
```
Comment (akhabali) [inline main/plugins/org.talend.sdk.component.studio-integration/messages.properties]: This error message is useless

**Comment by cmeng-talend** [comment]
I think we can't judge guess schema succeed or not though empty status of error stream, cause we can't make sure users' component won't print anything into error stream during their component execution, we should judge guess schema succeed or not based on the execution result instead of the error stream:
If returnCode == 0, means guess schema succeed, we can try to get result from out stream;
If returnCode != 0, means guess schema failed, we can read the error stream and show them to users.

**Comment by cmeng-talend** [comment]
@akhabali how do you think?

**Comment by rmannibucau** [comment]
Guess the opposite works better: if you got expected json line - we can ignore others - from stdout then all good, else read stderr and show it all. Exit code = 0 does not help as you explained.

**Comment by cmeng-talend** [comment]
@rmannibucau updated

**Comment by cmeng-talend** [comment]
@rmannibucau fix updated, since we need to check all lines starting with '[' and ending with ']', and there may be risk those lines are logs printed by user's component, so we have to get those lines and error stream at the same time when resolving metadata from those lines, if no valid metadata resolved, then means maybe guess schema failed, will print all messages from error stream.

File: main/plugins/org.talend.sdk.component.studio-integration/messages.properties (hunk 1 above)
Comment (cmeng-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/messages.properties]: will still print it if no any error returned from guess schema job, it's more friendly

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ui/guessschema/GuessSchemaSelectionAdapter.java [hunk 2]
```diff
@@ -179,31 +197,42 @@ public void widgetSelected(final SelectionEvent event) {
     }
 
     private IMetadataTable buildMetadata(final String schema) {
-        if (schema.trim().isEmpty()) {
+        if (schema == null || schema.trim().isEmpty()) {
             return null;
         }
+        Collection<MetadataColumn> jsonColumns = new HashSet<>();
+        final Jsonb jsonb = JsonbBuilder.create();
```
Comment (rmannibucau) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ui/guessschema/GuessSchemaSelectionAdapter.java]: should be a try with close resource as it was before

Comment (cmeng-talend) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/ui/guessschema/GuessSchemaSelectionAdapter.java]: @rmannibucau ok, updated fix, thanks

**Comment by cmeng-talend** [comment]
Hello @rmannibucau  @akhabali , does this PR have other issue? if there is no issue, could we merge it now? we need to merge this fix and test it as soon as possible for the release.

**Pull request merged**